tests/unit/test_endpoint_provider.py

```diff
@@ -152,6 +152,22 @@ def test_provider_unit_sets_bind_address_on_pebble_ready(self, *unused):
         self.assertIn("prometheus_scrape_unit_address", data)
         self.assertEqual(data["prometheus_scrape_unit_address"], "192.0.8.2")
 
+    @patch_network_get(private_address="192.0.8.2")
+    def test_provider_unit_sets_bind_address_on_update_status(self, *unused):
+        rel_id = self.harness.add_relation(RELATION_NAME, "provider")
+        self.harness.charm.on.update_status.emit()
+        data = self.harness.get_relation_data(rel_id, self.harness.charm.unit.name)
+        self.assertIn("prometheus_scrape_unit_address", data)
+        self.assertEqual(data["prometheus_scrape_unit_address"], "192.0.8.2")
+
+    @patch_network_get(private_address=None)
+    def test_provider_does_not_set_invalid_unit_address(self, *unused):
```

Comment:
Is the function name up to date? It says "not invalid".

Reply:
It says provider *does not set* invalid unit address. Am I missing something ?

Reply:
I'm just confused:
- There is `private_address=None` and `assertNotIn("prometheus_scrape_unit_address" ...`, so it is invalid
- If it is not invalid, then why not call it "provider sets valid address"?

Reply:
The test is to ensure that `MetricsEndpointProvider` does not set `prometheus_scrape_unit_address` if it gets an invalid (eg. `None`) unit address. So `MetricsEndpointProvider` is giving a guarantee that if it sets a unit IP address, it will be valid.